### ISSUE TITLE
Improved edit prompt for Gemini Pro 1.5, other Gemini models

### ISSUE THREAD

**Issue by sestinj** [continuedev/continue#3304]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

Gemini Pro 1.5 can sometimes repeat the entire file when the current Edit prompt is used.

### Solution

 This just needs some model-specific experimentation to figure out what instructions allow it to reliably give the proper output format. The edit prompts are found in https://github.com/continuedev/continue/blob/95f663b98f2ce5f73cee3f84d434a9da557788ae/core/llm/templates/edit.ts

**Comment by kevinsimper** [comment]
Gemini outputs this which makes inline editing not really useful unfortunately
![image](https://github.com/user-attachments/assets/4a4e68c8-5234-4723-a141-8c331e509aa1)

**Comment by sestinj** [comment]
Yup, exactly why we need to revisit the prompt, but once we adjust it I think everything will work great!
